rdmo/projects/progress.py

```diff
@@ -47,6 +47,44 @@ def compute_sets(values):
     return sets
 
 
+def compute_next_relevant_page(current_page, direction, catalog, resolved_conditions):
+    # compute the next relevant page based on resolved conditions.
+    # If no current_page, return None
+    if not current_page:
```

Comment:
since we always start with an existing page, we could move this breaking of the recursion at the end of the function